src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -300,6 +301,10 @@ private void drawCurvedPath(Graphics2D graphics, Point[] points, int xOffset, in
         new Line2D.Double(new Point2D.Double(xcoords[xcoords.length - 1], ycoords[ycoords.length - 1]),
             points[points.length - 1]),
         xOffset, yOffset, scale);
+    if (points[points.length - 2].distance(points[points.length - 1]) > 4) {
```

Comment:
Magic number 4.
Should be a constant, and use it to calculate the arrow size, or vice versa. Namely updating the arrow size should update this automatically. I assume the `>4` is to ensure there is enough space to draw the error and a pixel extra. 

Have you tested this to get to the `4` value, or is it logically a good one to start with? If the latter, it might make sense to require at least 2~3 pixels to draw a line before an arrow starts to look right. I plan to test this at some point soon, but for the moment I wonder what the testing results look like on short lines